src/components/legend/draw.js

```diff
@@ -392,24 +393,57 @@ function drawTexts(g, gd) {
                 this.text(text)
                     .call(textLayout);
 
+                var origText = text;
+
                 if(!this.text()) text = ' \u0020\u0020 ';
 
-                var fullInput = legendItem.trace._fullInput || {},
-                    astr;
+                var i, transforms, direction;
+                var fullInput = legendItem.trace._fullInput || {};
+                var needsRedraw = false;
+                var update = {};
 
                 // N.B. this block isn't super clean,
                 // is unfortunately untested at the moment,
                 // and only works for for 'ohlc' and 'candlestick',
                 // but should be generalized for other one-to-many transforms
                 if(['ohlc', 'candlestick'].indexOf(fullInput.type) !== -1) {
-                    var transforms = legendItem.trace.transforms,
-                        direction = transforms[transforms.length - 1].direction;
+                    transforms = legendItem.trace.transforms;
+                    direction = transforms[transforms.length - 1].direction;
+
+                    update[direction + '.name'] = text;
+                } else if(fullInput.transforms) {
+                    for(i = fullInput.transforms.length - 1; i >= 0; i--) {
+                        if(fullInput.transforms[i].type === 'groupby') {
```

Comment:
Might be nice to add a lib function similar to [`Registry.traceIs`](https://github.com/plotly/plotly.js/blob/b2e54861dfe9960a7bb447f0a622d4198a4e22c7/src/registry.js#L150) for transforms e.g. `Registry,hasTransform('groupby')`